index.js

```diff
@@ -67,6 +67,42 @@ function toggleColor(element) {
     }
 }
 
+header.addEventListener("click", function() {
+    toggleColor(header)
+})
+
+
+
 /***** Deliverable 2 *****/
 
+const button = document.querySelector('button#like-button')
+
+    button.addEventListener('click', function() {
+
+    console.log("clicked")
+
+    const totalLikes =  parseInt(likes.textContent)
+    likes.textContent = totalLikes + 1 + " Likes"
+})
+
+
 /***** Deliverable 3 *****/
+
+const newAnimalForm = document.querySelector("#new-animal-sighting-form")
+
+newAnimalForm.addEventListener('submit', function (event) {
+    event.preventDefault() 
+
+    console.log('submit!')
+
+    const newSpecies = event.target[0].value
+    const newVideo = event.target[1].value
+    const newPhoto = event.target[2].value
+    const newDescription = event.target[3].value
+
+    const newAnimalSighting = {
+
+        ///ran out of time.. :(
+    }
```

Comment:
Very close!! Again, I would create a separate function to be passed in as a callback function. I love the descriptive variable names!

Good start on assembling the data object!


My steps for solving this are as follows:

1. Find the sightingForm on the page and save it to a variable
2. add an event listener to the form element, listen for a `submit` event.
3. The submit event needs to be applied to an entire form element, not the submit button.
4. Create the `handleFormSubmission` callback function, which should be triggered when the form submits.

`handleFormSubmission` steps:

1. since we are submitting a form, we need to prevent the default action of the form, which is to make a post request and redirect to a specific URL. We do not want the page to refresh, we want to prevent that action and stay on the current page.
2. save all the data the user typed in from the form into descriptive variables.
3. assemble an object with key value pairs descriptive of the data they contain.
4. Pass in the animal data object to the `renderAnimalSightingPost` function

```
const sightingForm = document.querySelector(`form#new-animal-sighting-form`);

function handleFormSubmission(e){
    e.preventDefault();
    const species = e.target["species"].value;
    const videoLink = e.target["animal-link"].value;
    const photo = e.target["animal-photo"].value;
    const description = e.target["sighting-description"].value;
    const newData = {
        species: species,
        videoLink: videoLink,
        photo: photo,
        description: description
    }
    renderAnimalSightingPost(newData);
}

sightingForm.addEventListener("submit", handleFormSubmission)
```

With ES6 notation

```
const sightingForm = document.querySelector(`form#new-animal-sighting-form`);

const handleFormSubmission = e => {
    e.preventDefault();
    const species = e.target["species"].value;
    const videoLink = e.target["animal-link"].value;
    const photo = e.target["animal-photo"].value;
    const description = e.target["sighting-description"].value;
    const newData = {
        species,
        videoLink,
        photo,
        description
    }
    e.target.reset()
    renderAnimalSightingPost(newData);
}

sightingForm.addEventListener("submit", handleFormSubmission)
```